ctru-rs/src/services/ndsp/mod.rs

```diff
@@ -0,0 +1,243 @@
+pub mod wave;
+use wave::{WaveInfo, WaveStatus};
+
+use crate::error::ResultCode;
+use crate::services::ServiceReference;
+
+use once_cell::sync::Lazy;
+use std::sync::Mutex;
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum OutputMode {
+    Mono = ctru_sys::NDSP_OUTPUT_MONO,
+    Stereo = ctru_sys::NDSP_OUTPUT_STEREO,
+    Surround = ctru_sys::NDSP_OUTPUT_SURROUND,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum AudioFormat {
+    PCM8Mono = ctru_sys::NDSP_FORMAT_MONO_PCM8,
+    PCM16Mono = ctru_sys::NDSP_FORMAT_MONO_PCM16,
+    ADPCMMono = ctru_sys::NDSP_FORMAT_MONO_ADPCM,
+    PCM8Stereo = ctru_sys::NDSP_FORMAT_STEREO_PCM8,
+    PCM16Stereo = ctru_sys::NDSP_FORMAT_STEREO_PCM16,
+    FrontBypass = ctru_sys::NDSP_FRONT_BYPASS,
+    SurroundPreprocessed = ctru_sys::NDSP_3D_SURROUND_PREPROCESSED,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[repr(u32)]
+pub enum InterpolationType {
+    Polyphase = ctru_sys::NDSP_INTERP_POLYPHASE,
+    Linear = ctru_sys::NDSP_INTERP_LINEAR,
+    None = ctru_sys::NDSP_INTERP_NONE,
+}
+
+pub struct Channel {
+    id: i32,
+}
+
+static NDSP_ACTIVE: Lazy<Mutex<usize>> = Lazy::new(|| Mutex::new(0));
```

Comment:
`Mutex::new` is now const, no `Lazy` required :)

Reply:
I missed this! We have some cleanup to do… 😄